osf/models/files.py

```diff
@@ -445,7 +445,7 @@ def update(self, revision, data, user=None):
                 break
         else:
             # Insert into history if there is no matching etag
-            if data.get('modified'):
+            if data.get('modified') and None not in [item['modified'] for item in self.history]:
```

Comment:
@adlius, sorry I'm conflating two different issues.

1. The bit about adding a debug statement would be for local dev only and not a part of the PR.  I'm talking about adding a `logging.error` call to the `update()` method, running the full test suite, then seeing how many times it gets called and from where.  That'll help us figure out where `update()` is being called and what its expected behavior is there.  If all the calls come about as a result of `touch()`, then this fix is probably safe.  If there are other places that call `update()`, we need to see if they require the sorting behavior or if append-only works for them.  If they require the sorting behavior, the fix for this will need to be more complicated.  This issue is answering the question *"will switching to append-only break the non-touch parts of the OSF that call `File.update()`*.

2. The second issue is the sentry logging message.  I'm pretty sure that `update()` called from `touch()` will never receive metadata that is older than the the newest entry already in history.  However, it would be nice to check this to make sure I'm not forgetting some use case.  Therefore, I'd like a sentry message that warns if the new revision information is older than the most recent revision.  This issue is answering the question *"is it correct to assume that metadata from the touch() method will always be the same age or newer than whatever we've stored already"*.

PLMK if I'm talking gibberish.